server/service/service_users_test.go

```diff
@@ -591,28 +605,31 @@ func TestPerformRequiredPasswordReset(t *testing.T) {
 			_, err = svc.RequirePasswordReset(test.UserContext(test.UserAdmin), user.ID, true)
 			require.Nil(t, err)
 
+			ctx = refreshCtx(t, ctx, user, ds, nil)
+
 			// should error when not logged in
-			_, err = svc.PerformRequiredPasswordReset(test.UserContext(test.UserAdmin), "new_pass")
-			require.NotNil(t, err)
+			// FIXME: we don't check for session anymore... should we?
```

Comment:
Can you elaborate on this? Might be nice to fix this while we are here.

Reply:
The code doesn't seem to validate sessions anywhere, the current user is gotten from the ctx, which in this case is an admin as generated by `test.UserContext`. When we don't have the rest of the layers, the current user seems to always be taken from the context and Viewer.Session is never really used within this scope.

I might be missing something, though, it can get dizzying to go back and forth with the auth checks

Reply:
During a request it should only be possible to get a user into the context if they have a valid session, so I'm not sure that we need to do any checking of the session for this method. Does that seem reasonable to you?

Reply:
Makes sense. We should add a test at that level for this if it doesn't have one.